include/audio/codec.h

```diff
@@ -0,0 +1,215 @@
+} audio_dai_type_t;
+
+/**
+ * Codec properties that can be set by audio_codec_set_property()
+ */
+typedef enum {
+	AUDIO_PROPERTY_OUTPUT_VOLUME,
+	AUDIO_PROPERTY_OUTPUT_MUTE,
+} audio_property_t;
+
+/**
+ * Audio channel identifiers to use in audio_codec_set_property()
+ */
+typedef enum {
+	AUDIO_CHANNEL_FRONT_LEFT,
+	AUDIO_CHANNEL_FRONT_RIGHT,
+	AUDIO_CHANNEL_LFE,
+	AUDIO_CHANNEL_FRONT_CENTER,
+	AUDIO_CHANNEL_REAR_LEFT,
+	AUDIO_CHANNEL_REAR_RIGHT,
+	AUDIO_CHANNEL_REAR_CENTER,
+	AUDIO_CHANNEL_SIDE_LEFT,
+	AUDIO_CHANNEL_SIDE_RIGHT,
+	AUDIO_CHANNEL_ALL,
+} audio_channel_t;
+
+/**
+ * Digital Audio Interface Configuration
+ * Configuration is dependent on DAI type
+ */
+typedef union {
+	struct i2s_config i2s;	/* I2S configuration */
+				/* Other DAI types go here */
+} audio_dai_cfg_t;
+
+/**
+ * Codec configuration parameters
+ */
+struct audio_codec_cfg {
+	u32_t			mclk_freq;	/* MCLK input frequency in Hz */
+	audio_dai_type_t	dai_type;	/* Digital interface type */
+	audio_dai_cfg_t		dai_cfg;	/* DAI configuration info */
+};
+
+/**
+ * Codec property values
+ */
+typedef union {
+	int 	vol;	/* Volume level in 0.5dB resolution */
+	bool 	mute;	/* mute if true, unmute if false */
+} audio_property_value_t;
+
+/**
+ * @cond INTERNAL_HIDDEN
+ *
+ * For internal use only, skip these in public documentation.
+ */
+struct audio_codec_api {
+	int (*configure)(struct device *dev, struct audio_codec_cfg *cfg);
+	void (*start_output)(struct device *dev);
+	void (*stop_output)(struct device *dev);
+	int (*set_property)(struct device *dev, audio_property_t property,
+			audio_channel_t channel, audio_property_value_t val);
+	int (*apply_properties)(struct device *dev);
+};
+/**
+ * @endcond
+ */
+
+/**
+ * @brief Configure the audio codec
+ *
+ * Configure the audio codec device according to the configuration
+ * parameters provided as input
+ *
+ * @param dev Pointer to the device structure for codec driver instance.
+ * @param cfg Pointer to the structure containing the codec configuration.
+ *
+ * @return 0 on success, negative error code on failure
+ */
+static inline int audio_codec_configure(struct device *dev,
+		struct audio_codec_cfg *cfg)
+{
+	const struct audio_codec_api *api = dev->driver_api;
+
+	return api->configure(dev, cfg);
+}
+
+/**
+ * @brief Set codec to start output audio playback
+ *
+ * Setup the audio codec device to start the audio playback
+ *
+ * @param dev Pointer to the device structure for codec driver instance.
+ *
+ * @return none
+ */
+static inline void audio_codec_start_output(struct device *dev)
+{
+	const struct audio_codec_api *api = dev->driver_api;
+
+	api->start_output(dev);
+}
+
+/**
+ * @brief Set codec to stop output audio playback
+ *
+ * Setup the audio codec device to stop the audio playback
+ *
+ * @param dev Pointer to the device structure for codec driver instance.
+ *
+ * @return none
+ */
+void audio_codec_stop_output(struct device *dev)
```

Comment:
static inline ？

Reply:
> static inline ？

good catch. will fix this as well as `audio_codec_set_property` and `audio_codec_apply_properties`